base/deprecated.jl

```diff
@@ -1581,6 +1581,10 @@ end
 
 @deprecate String(io::GenericIOBuffer) String(take!(copy(io)))
 
+
+@deprecate readstring(s::IO) read(s, String)
+@deprecate readstring(filename::String) read(filename, String)
```

Comment:
This seems to be tighter than the [removed method](https://github.com/JuliaLang/julia/pull/22864/files#diff-775ff60d66216b33269fc5fcb6be714fL536), should be `AbstractString`?